pandas/core/frame.py

```diff
@@ -3354,7 +3354,7 @@ def dropna(self, axis=0, how='any', thresh=None, subset=None,
                 else:
                     raise TypeError('must specify how or thresh')
 
-            result = self.take(mask.nonzero()[0], axis=axis, convert=False)
```

Comment:
I don't think these internal uses should be removed.  We're guaranteed the indexer is inbounds, so it's an unnecessary performance hit to check it.

Reply:
What do you mean by this?  `DataFrame.take` doesn't even respect the `convert` parameter.  Thus, this will have no effect on anything. 😄

Reply:
Ok, this example doesn't matter b/c it's on a frame, but does with the usages of  `_data.take`

Reply:
And `Series.take`

Reply:
I don't think there's a bounds check anymore in `Series.take` or `_data.take` because I removed all of them for that same reason.  We have so many bounds checks all over the place that removing just one of them is okay now.